### DISCUSSION TITLE
Can INSTALL-CHOCOLATEYPACKAGE do xcopy for the exe file in toolsDir.

### DISCUSSION THREAD

**Discussion by moh-hassan** [chocolatey/choco#2461] [Q&A]
I have a package with script 'chocolateyInstall.ps1'  that call the command 'INSTALL-CHOCOLATEYPACKAGE' and download remote URL of exe file (not  a setup).
Although the installation is successful, the package is not ready to run because the exe is not available in the toolsDir.
Install-ChocolateyPackage command download the file in a temporary location but didn't copy the file to the toolsDir.

I have to zip the file and use the command 'Install-ChocolateyZipPackage' which copy the exe to  toolsDir and is working fine.

It's nice if the command 'INSTALL-CHOCOLATEYPACKAGE'  can do xcopy of  the  exe file(s) as if it's portable package without imbedding the exe in the package.

**Comment by TheCakeIsNaOH** [comment]
@moh-hassan You can use `Get-ChocolateyWebFile` to download the exe file directly to the `$toolsDir`

https://docs.chocolatey.org/en-us/create/functions/get-chocolateywebfile

**Comment by moh-hassan** [comment]
Thanks @TheCakeIsNaOH  for reply.
The package will be published on chocolatey community . I may not have  a permission in the file system to download the remote exe in a $toolsdir.
Kindly, Can you show me use case for a published  package using  Get-ChocolateyWebFile.

**Comment by TheCakeIsNaOH** [comment]
> I may not have a permission in the file system to download the remote exe in a $toolsdir.

If you have a normal installation of Chocolatey, then choco should be run with administrative privileges, in which case choco would have write permissions into the `$toolsDir`

If you have a user level installation of Chocolatey, then the `$toolsDir` should be writable by your user.

> Can you show me use case for a published package using Get-ChocolateyWebFile.

Here is an example of a package downloading an exe with `Get-ChocolateyWebFile`

https://github.com/chocolatey-community/chocolatey-packages/blob/master/automatic/win10mct/tools/chocolateyInstall.ps1

**Comment by moh-hassan** [comment]
I used the script you guided me, and the package is installed successfully. 👍 
I noticed that chocolatey create two copies in different locations  with different hash256:
C:\ProgramData\chocolatey\bin\myconsole.exe                                 #shimed
C:\ProgramData\chocolatey\lib\mypackage\tools\myconsole.exe     #origeinal fil

with the message:
```
shimGen has successfully created a shim for myconsole.exe
  Created: C:\ProgramData\chocolatey\bin\myconsole.exe
  Targeting: C:\ProgramData\chocolatey\lib\odata2poco-commandline\tools\myconsole.exe
  IsGui:False

 The install of mypackage was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.
  
```
 Thanks for support

**Comment by TheCakeIsNaOH** [comment]
Here is information about shims:
https://docs.chocolatey.org/en-us/features/shim

**Comment by moh-hassan** [comment]
Now, it's clear why two files.

**Comment by pauby** [comment]
This isn't an issue with Chocolatey, so I've moved it over to a discussion. If that changes, we can always move it back.
